### PR TITLE
Make Sessions resilient to the FirebaseApp instance being deleted

### PR DESCRIPTION
**Pull request by mrober**

### CODE REVIEW

**Comment by mrober** [comment]
#5186

File: firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerformanceInitializer.java [hunk 1]
```diff
@@ -15,17 +15,23 @@
...
+  private static final AndroidLogger logger = AndroidLogger.getInstance();
 
   @Override
   public void onAppColdStart() {
     // Initialize FirebasePerformance when app cold starts.
-    FirebasePerformance.getInstance();
+    try {
+      FirebasePerformance.getInstance();
+    } catch (IllegalStateException ex) {
+      logger.info("Failed to get instance of FirebasePerformance: %s", ex);
```
Comment (visumickey) [inline firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerformanceInitializer.java:34]: Do we want to give some more verbose message here? Eg: `Firebase Performance not initialized. We would not be collecting any performance metrics. Make sure FirebaseApp is initialized.`

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt [hunk 2]
```diff
@@ -138,9 +138,14 @@ internal constructor(
...
-    sessionCoordinator.attemptLoggingSessionEvent(
-      SessionEvents.startSession(firebaseApp, sessionDetails, sessionSettings, subscribers)
-    )
+    try {
+      val sessionEvent =
+        SessionEvents.startSession(firebaseApp, sessionDetails, sessionSettings, subscribers)
+      sessionCoordinator.attemptLoggingSessionEvent(sessionEvent)
+    } catch (ex: IllegalStateException) {
+      // This can happen if the app suddenly deletes the instance of FirebaseApp.
+      Log.i(TAG, "Sessions SDK failed to log Session Start event.", ex)
```
Comment (visumickey) [inline firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt:147]: Let's make this log message a bit more verbose as well. Basically, we want to signal to the developer that unregistering a FirebaseApp would not collect any data.

File: firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerformanceInitializer.java [hunk 3]
```diff
@@ -15,17 +15,26 @@
...
+  private static final AndroidLogger logger = AndroidLogger.getInstance();
 
   @Override
   public void onAppColdStart() {
     // Initialize FirebasePerformance when app cold starts.
-    FirebasePerformance.getInstance();
+    try {
+      FirebasePerformance.getInstance();
+    } catch (IllegalStateException ex) {
+      logger.info(
```
Comment (visumickey) [inline firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerformanceInitializer.java:34]: Should we elevate the log level to `Warning`?

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt [hunk 4]
```diff
@@ -138,9 +143,18 @@ internal constructor(
...
-    sessionCoordinator.attemptLoggingSessionEvent(
-      SessionEvents.startSession(firebaseApp, sessionDetails, sessionSettings, subscribers)
-    )
+    try {
+      val sessionEvent =
+        SessionEvents.startSession(firebaseApp, sessionDetails, sessionSettings, subscribers)
+      sessionCoordinator.attemptLoggingSessionEvent(sessionEvent)
+    } catch (ex: IllegalStateException) {
+      // This can happen if the app suddenly deletes the instance of FirebaseApp.
+      Log.i(
```
Comment (visumickey) [inline firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt:152]: `Info` or `Warning`? The previous log message is at `Warning`.

**Review by visumickey — APPROVED** [review]
Thanks for making the change.

File: firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerformanceInitializer.java (hunk 3 above)
Comment (mrober) [reply firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerformanceInitializer.java]: Made them all log as warnings

**Pull request merged**